### ISSUE TITLE
FATAL ERROR: v8::FromJust Maybe value is Nothing.

### ISSUE THREAD

**Issue by OmgImAlexis** [vitest-dev/vitest#2175]
### Describe the bug

Similar issue as https://github.com/vitest-dev/vitest/issues/1191

When hitting this I'm redirected to that issue but yet can't add a comment as it's "This conversation has been locked as resolved and limited to collaborators.".

```
FATAL ERROR: v8::FromJust Maybe value is Nothing.
 1: 0xb652e0 node::Abort() [node]
 2: 0xa761b5 node::FatalError(char const*, char const*) [node]
 3: 0xd5595a v8::Utils::ReportApiFailure(char const*, char const*) [node]
 4: 0xb6974d node::fs::FileHandle::CloseReq::Resolve() [node]
 5: 0xb698d0  [node]
 6: 0x16552fd  [node]
 7: 0x1659ad6  [node]
 8: 0x166c214  [node]
 9: 0x165a428 uv_run [node]
10: 0xaa1e45 node::SpinEventLoop(node::Environment*) [node]
11: 0xc33d32 node::worker::Worker::Run() [node]
12: 0xc342f8  [node]
13: 0x7f5d64a16ea7  [/lib/x86_64-linux-gnu/libpthread.so.0]
14: 0x7f5d64946def clone [/lib/x86_64-linux-gnu/libc.so.6]
```

While hitting this I also noticed sometimes my tests just stop without a crash. There's no summary, error, etc. they just stop. 
```
root@eae0f519711d:/app# npm run test

> @unraid/api@2.52.1 test
> vitest run --segfault-retry=3


 RUN  v0.23.4 /app

 ✓ src/__test__/test_name_here.test.ts (17) 593ms
 ✓ src/__test__/another_test_name_here.test.ts (1) 5036ms
 ```

### Reproduction

I've yet to find what's causing this, it seems to happen randomly in my tests. If someone could point me to how to debug this I'm happy to try and get a repro going.

### System Info

```shell
System:
    OS: Linux 5.10 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
    CPU: (12) x64 Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz
    Memory: 6.41 GB / 7.72 GB
    Container: Yes
    Shell: 5.1.4 - /bin/bash
  Binaries:
    Node: 18.5.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.12.1 - /usr/local/bin/npm
  npmPackages:
    vitest: ^0.24.3 => 0.24.3
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
1) Stacktrace here is different than it is in the mention issue, so new issue is preferable here.
2) Have you tried updating your Node version to the latest? 
3) Cannot help you without reproduction or at least a project to debug, if it's open source.

**Comment by ivands** [comment]
I'm getting this error as well, using Node: v18.3.0

**Comment by sheremet-va** [comment]
> I'm getting this error as well, using Node: v18.3.0

Node 18.3 is not the latest Node 18

**Comment by Priestch** [comment]
@OmgImAlexis I got a similar error in a test file, it uses the fs module to sync read, and write to some files. 

My OS is Ubuntu 22.04, after upgrading vitetest from v0.27.1 to v0.28.3, the error just disappeared, hope this helps.

**Comment by revers3ntropy** [comment]
I got this on v0.33.0 and downgrading to v0.32.4 fixed it...

**Comment by VKeyPEDIdea** [comment]
> I got this on v0.33.0 and downgrading to v0.32.4 fixed it...

I got this issue on v0.34.1 and v0.34.4. So I downgraded this to v0.32.4. After it I can't get the error.

**Comment by pezillionaire** [comment]
Similar Issues, I noticed that removing `junit` from my reporters seems to resolve this on v1.0.1

**Comment by liangskyli** [comment]
Same issue, version: vitest@1.3.1

**Comment by AriPerkkio** [comment]
Fixed by https://github.com/vitest-dev/vitest/pull/3925 and https://github.com/vitest-dev/vitest/pull/4172.

Vitest 1.0.0 has now `--pool=forks` which uses multiple `node:child_process` in parallel. The previous `--no-threads` was using a single `node:child_process`.

If the code you are testing is incompatible with `node:worker_threads`, switch to `--pool=forks`. If you run into same error with that pool, feel free to open new issue with minimal reproduction.
